alvr/common/src/commands.rs

```diff
@@ -228,45 +228,57 @@ fn netsh_delete_rule_command_string(rule_name: &str) -> String {
 pub fn firewall_rules(add: bool) -> Result<(), i32> {
-    let script_path = env::temp_dir().join("alvr_firewall_rules.bat");
-
-    let firewall_rules_script_content = if add {
-        format!(
-            "{}\n{}",
-            netsh_add_rule_command_string(
-                "SteamVR ALVR vrserver",
-                &steamvr_root_dir()
-                    .map_err(|_| -1)?
-                    .join("bin")
-                    .join("win64")
-                    .join("vrserver.exe")
-            ),
-            netsh_add_rule_command_string(
-                "SteamVR ALVR vrserver",
-                &steamvr_root_dir()
-                    .map_err(|_| -1)?
-                    .join("bin")
-                    .join("win32")
-                    .join("vrserver.exe")
-            ),
-        )
+    let exit_status;
+
+    if cfg!(target_os = "linux") {
+        let action = if add { "add" } else { "remove" };
+        // run as normal user since we use pkexec to sudo
+        exit_status = Command::new("bash")
+            .arg("/usr/libexec/alvr/alvr_fw_config.sh")
+            .arg(action)
+            .status()
+            .map_err(|_| -1)?;
     } else {
-        format!(
-            "{}\n{}",
-            netsh_delete_rule_command_string("ALVR Launcher"),
-            netsh_delete_rule_command_string("SteamVR ALVR vrserver")
-        )
-    };
-    fs::write(&script_path, firewall_rules_script_content).map_err(|_| -1)?;
-
-    // run with admin priviles
-    let exit_status = runas::Command::new(script_path)
-        .show(cfg!(target_os = "linux"))
-        .gui(true) // UAC, if available
-        .status()
-        .map_err(|_| -1)?;
+        let script_path = env::temp_dir().join("alvr_firewall_rules.bat");
+        let firewall_rules_script_content = if add {
+            format!(
+                "{}\n{}",
+                netsh_add_rule_command_string(
+                    "SteamVR ALVR vrserver",
+                    &steamvr_root_dir()
+                        .map_err(|_| -1)?
+                        .join("bin")
+                        .join("win64")
+                        .join("vrserver.exe")
+                ),
+                netsh_add_rule_command_string(
+                    "SteamVR ALVR vrserver",
+                    &steamvr_root_dir()
+                        .map_err(|_| -1)?
+                        .join("bin")
+                        .join("win32")
+                        .join("vrserver.exe")
+                ),
+            )
+        } else {
+            format!(
+                "{}\n{}",
+                netsh_delete_rule_command_string("ALVR Launcher"),
+                netsh_delete_rule_command_string("SteamVR ALVR vrserver")
+            )
+        };
+        fs::write(&script_path, firewall_rules_script_content).map_err(|_| -1)?;
+
+        // run with admin priviles
+        exit_status = runas::Command::new(script_path)
+            .show(cfg!(target_os = "linux"))
```

Comment:
Unless the GitHub viewer is messing with my brain, this will always be false since we are inside the non-linux branch.